Stratis.Bitcoin/Utilities/AsyncLock.cs

```diff
@@ -1,131 +1,57 @@
 ﻿using System;
-using System.Collections;
 using System.Collections.Generic;
-using System.Collections.ObjectModel;
-using System.Linq;
+using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
-using NBitcoin;
 
 namespace Stratis.Bitcoin.Utilities
 {
     /// <summary>
-    /// An async reader writer lock for concurrent and exclusive work.
+    /// Usage: using(await myAsyncLock.LockAsync())
+    /// Implementation: https://www.hanselman.com/blog/ComparingTwoTechniquesInNETAsynchronousCoordinationPrimitives.aspx
     /// </summary>
```

Comment:
sorry, this is not enough - please see https://github.com/stratisproject/StratisBitcoinFullNode/blob/master/Documentation/commenting-policy.md

The idea is to have every piece commented and using correct XML tags. In summary you describe the thing. Usage belongs to different section and where you copied it from belongs to seealso section. 

fields and methods should be commented, the comments should not be external (on different website) they should be in the source code file